### PR TITLE
Exponent

### PR DESCRIPTION
**Pull request by 404Naz**
Completed exponent for variable base and real power and real base and real power. Addresses #1

### CODE REVIEW

File: include/Oasis/Exponent.hpp
Comment (matthew-mccall) [inline include/Oasis/Exponent.hpp:1]: Make sure to add Doxygen comments: https://www.doxygen.nl/manual/docblocks.html

**Review by matthew-mccall — CHANGES_REQUESTED** [review]
We have to work on replacing the specific cases for `Variable` with cases that work on any general `Expression`

**Review by Jdpankow — APPROVED** [review]

File: src/Add.cpp
Comment (matthew-mccall) [inline src/Add.cpp:1]: You need to make for this work for any `Expression` not just `Variable`

File: src/Subtract.cpp
Comment (matthew-mccall) [inline src/Subtract.cpp:1]: You also should update this

**Review by matthew-mccall — CHANGES_REQUESTED** [review]
In `Add` and `Subtract` you still need to update the code to handle and `Expression` instead of just `Variable`s

**Comment by 404Naz** [comment]
Updated Add.cpp and Subtract.cpp

**Review by matthew-mccall — APPROVED** [review]

**Pull request merged**